extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLAbstractHandler.java

```diff
@@ -56,11 +63,16 @@ public void handle(final RoutingContext ctx) {
         } else {
 
             currentManagedContext.activate();
-            handleWithIdentity(ctx);
-
-            ctx.response().bodyEndHandler((e) -> {
+            ctx.response()
+                    .endHandler(currentManagedContextTerminationHandler)
+                    .exceptionHandler(currentManagedContextTerminationHandler)
+                    .closeHandler(currentManagedContextTerminationHandler);
+
+            try {
+                handleWithIdentity(ctx);
+            } catch (Throwable t) {
                 currentManagedContext.terminate();
```

Comment:
I'm not like totally exactly 1000% sure, but ... shouldn't we rethrow the exception? Or there's nothing above this handler that could meaningfully handle it and swallowing is fine?

Reply:
You are totally right! 

@phillip-kruger can you open another PR?

Reply:
I did something here as I would like it to be merged tonight: https://github.com/quarkusio/quarkus/pull/26797